rest/data/commit_queue.go

```diff
@@ -385,15 +385,27 @@ func checkPRApprovals(ctx context.Context, settings *evergreen.Settings, userRep
 		return errors.Wrap(err, "getting GitHub OAuth token from settings")
 	}
 
-	reviews, err := thirdparty.GetGithubPullRequestReviews(ctx, githubToken, userRepo.Owner, userRepo.Repo, prNum)
-	if err != nil {
-		return errors.Wrap(err, "getting GitHub PR reviews")
-	}
-
 	var numApprovals int
-	for _, r := range reviews {
-		if r.GetState() == githubReviewApproved {
-			numApprovals += 1
+	reviewsPage := 0
+	var reviews []*github.PullRequestReview
+
+	for numApprovals < requiredApprovalCount {
```

Comment:
Can we turn these nested for loops into a helper like `getNumApprovals(requiredApprovalCount)` that simply returns the number of approvals that it found (up to the required amount)? Would avoid having to think about the for loop `break`'ing behavior.

Reply:
It seems to me like that would just move this for loop to another function, I don't know if that would have much of an impact on readability.

Reply:
Eh, I don't have that strong of opinions here. Generally, I try to break up double loops that use breaks/continues into helper functions (at least, put the inner loop in its own little function) because it avoids having to think about "am I breaking out of the inner loop or outer loop, does it still work if I just break out of the inner loop" and such. I don't think it's a big deal in this case.